src/testup/minitest_setup.rb

```diff
@@ -18,73 +18,55 @@
-        lapsed_time = Time.now - start_time
-        TestUp::Debugger.output("> Elapsed time: #{lapsed_time}s")
-        result
-      end
+module TestUp
+  module MinitestPrepend
+    def run_one_method(klass, method_name)
+      TestUp::Debugger.output("Running: #{klass.name}.#{method_name}")
+      start_time = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+      result = super
+      elapsed_time = Process.clock_gettime(Process::CLOCK_MONOTONIC) - start_time
+      TestUp::Debugger.output("> Elapsed time: #{'%.5f' % elapsed_time}s")
+      result
     end
-
-  end # class << self
-
-  class Runnable
-    class << self
-      unless method_defined?(:testup_run)
-        alias :testup_run :run
-        # This is mostly a copy+paste of Minitest::Runnable.run method from v5.4.3.
-        # It adds hooks called before and after a testcase is run.
-        # It's to allow setup or tests that might be slow to setup.
-        def run(reporter, options = {})
-          filter = options[:filter] || '/./'
-          filter = Regexp.new $1 if filter =~ /\/(.*)\//
-
-          filtered_methods = self.runnable_methods.find_all { |m|
-            filter === m || filter === "#{self}##{m}"
-          }
-
-          with_info_handler reporter do
-            # Custom TestUp hook
-            if !filtered_methods.empty? && self.respond_to?(:setup_testcase)
-              TestUp::Debugger.output("setup_testcase: #{self.name}")
-              self.setup_testcase
-            end
-
-            filtered_methods.each do |method_name|
-              run_one_method self, method_name, reporter
-            end
-
-            # Custom TestUp hook
-            if !filtered_methods.empty? && self.respond_to?(:teardown_testcase)
-              TestUp::Debugger.output("teardown_testcase: #{self.name}")
-              self.teardown_testcase
-            end
-          end
-        end # def run
+  end
+
+  module RunnablePrepend
+    def run(reporter, options = {})
+      filter = options[:filter] || '/./'
+      filter = Regexp.new $1 if filter =~ /\/(.*)\//
+
+      filtered_methods = runnable_methods.find_all { |m|
+        filter === m || filter === "#{self}##{m}"
+      }
+
+      with_info_handler reporter do
+        # Custom TestUp hook
+        if !filtered_methods.empty? && self.respond_to?(:setup_testcase)
+          TestUp::Debugger.output("setup_testcase: #{self.name}")
+          setup_testcase
+        end
+
+        filtered_methods.each do |method_name|
+          run_one_method self, method_name, reporter
+        end
+
+        # Custom TestUp hook
+        if !filtered_methods.empty? && self.respond_to?(:teardown_testcase)
+          TestUp::Debugger.output("teardown_testcase: #{self.name}")
+          teardown_testcase
+        end
       end
-    end # class << self
-  end # class Runnable
+    end # def run
+  end
+end
+
+# TestUp modifications of Minitest
+Minitest.singleton_class.prepend TestUp::MinitestPrepend
```

Comment:
Style: Use parentheses around arguments.

Reply:
Passes RuboCop.  If people aren't sure what it does, the Ruby docs do not use parentheses...